parse/parse.go

```diff
@@ -648,6 +674,10 @@ selectorsLoop:
 		case itemEOF:
 			return errorf("%w", unexpectedErr(itm))
 		case itemVariable:
+			if _, ok := p.annotations[itm.val]; !ok {
```

Comment:
Would it be better to compute direct/indirect annotations in the parseMatcher function instead? Otherwise, it computes annotations even when the message does not contain a matcher.